caracal/workers/selfcal_worker.py

```diff
@@ -561,8 +561,11 @@ def image(trg, num, img_dir, mslist, field):
             image_opts.update({
                 "auto-mask": config[key]['cleanmask_thr'][num-1 if len(config[key]['cleanmask_thr']) >= num else -1],
                 "local-rms": config[key]['cleanmask_localrms'][num-1 if len(config[key]['cleanmask_localrms']) >= num else -1],
-                "local-rms-window": config[key]['cleanmask_localrms_window'][num-1 if len(config[key]['cleanmask_localrms_window']) >= num else -1],
               })
+            if config[key]['cleanmask_localrms'][num-1 if len(config[key]['cleanmask_localrms']) >= num else -1]:
```

Comment:
This isjust cosmetics of course, but in the interest of readability (who will think of the future generations!), I would prefer to phrase that as 

```python
local_rms = config[key]['cleanmask_localrms'][min(num-1, len(config[key]['cleanmask_localrms'])-1)]
if local_rms:
    image_opts.update({
        "local-rms": local_rms, 
        "local-rms-window": config[key]['cleanmask_localrms_window'][min(num-1, len(config[key]['cleanmask_localrms'])-1)]
    })
```
...since that if-else construct is a big eyeful to take in, and recurs a lot.

Even better, define a helper function in there:

```
def n_1_or_last(seq, num):
    return seq[min(num-1, len(seq)-1)]
```

...since I presume this treatment of lists happens all over the place?

Reply:
yeah that would be neater -- we can sneak that into the next PR